src/main/java/de/tum/in/www1/bamboo/server/ServerNotificationTransport.java

```diff
@@ -252,11 +272,22 @@ private JSONObject createTestsResultsJSONObject(TestResults testResults, boolean
     }
 
     private JSONArray createTestsResultsJSONArray(Collection<TestResults> testResultsCollection, boolean addErrors) throws JSONException {
+        logToBuildLog("Creating test results JSON array");
         JSONArray testResultsArray = new JSONArray();
         for (TestResults testResults : testResultsCollection) {
             testResultsArray.put(createTestsResultsJSONObject(testResults, addErrors));
         }
 
         return testResultsArray;
     }
+
+    private void logToBuildLog(String s) {
+        BuildLogger buildLogger = buildLoggerManager.getLogger(plan.getPlanKey());
```

Comment:
I would add null checks here, in particular for `buildLoggerManager` and `plan` because they have the @Nullable annotation above.

`buildLogger` might also be null, so we should add another null check here!

Reply:
I've added the checks.